sips/sip-x.md

```diff
@@ -0,0 +1,67 @@
+---
+sip: <to be assigned>
+title: Minimal Proxies for Virtual Synths
+status: WIP
+author: Justin Moses (@justinjmoses), Brett Sun (@sohkai)
+discussions-to: https://research.synthetix.io/t/sip-x-minimal-proxies-for-virtual-synths/366
+
+created: 2021-03-31
+---
+
+## Simple Summary
+
+Deploy a ERC-1167 Minimal Proxy instead of a full Virtual Synth contract when new Virtual Synths are created.
+
+## Abstract
+
+This SIP proposes to optimize the creation of new Virtual Synths by deploying [ERC-1167 Minimal Proxies](https://eips.ethereum.org/EIPS/eip-1167) instead of the full [SIP-89 Virtual Synth](https://sips.synthetix.io/sips/sip-89) contract on each exchange outputting a Virtual Synth.
+
+## Motivation
+
+This SIP is motivated in two parts: users and code size limits (development headache).
+
+Users benefit from this optimization because ERC-1167 proxies are *tiny*. They are generally regarded as the cheapest way to deploy a copy of a contract and there are many copies of Virtual Synths needed (currently, a new one is deployed for the majority of exchanges on L1). Users should expect the gas cost of each exchange outputting Virtual Synths to fall to 800k gas (50% cheaper than current).
+
+Secondly, adopting ERC-1167 proxies reduces the code size of factory contracts that deploy contract copies, e.g. `ExchangerWithVirtualSynth`. A number of other such contracts in Synthetix are at or near the 24kb code size limit established in [EIP-170](https://eips.ethereum.org/EIPS/eip-170), hampering development as developers must struggle with optimizing new and existing code or significantly refactoring functionality to comply with the limit when any part of the dependency tree leading up to the factory is touched (e.g. [SIP-120](https://sips.synthetix.io/sips/sip-120)). Taking `ExchangerWithVirtualSynth` as an example, its code size will be reduced by 5kb, giving more wiggle room to fit SIP-120 and future changes in.
+
+Initially, only VirtualSynths are proposed to be deployed in this way as they represent the largest gas opportunity for users. In the future, other components such as binary options could also be updated.
+
+## Specification
+
+### Overview
+
+The `ExchangerWithVirtual` contract will be modified to:
+
+1. Deploy an ERC-1167 proxy pointing to an "base" Virtual Synth contract instead of a full Virtual Synth contract.
+2. Receive the address of the base Virtual Synth contract in its `constructor`. This value will be considered immutable with no functionality to update it in the future.
```

Comment:
for this, we can lookup the address of the static virtual synth into the `AddressResolver`